lightning/api_clnsocket.go

```diff
@@ -425,6 +429,259 @@ func (l *ClnSocketLightningAPI) GetNodeInfoFull(ctx context.Context, channels bo
 	return result, nil
 }
 
+// SubscribeForwards - API call
+func (l *ClnSocketLightningAPI) SubscribeForwards(ctx context.Context, since time.Time, batchSize uint16) (<-chan []ForwardingEvent, <-chan ErrorData) {
+	var reply ClnForwardEntries
+
+	if batchSize == 0 {
+		batchSize = 50
+	}
+
+	const maxErrors = 5
+
+	errors := 0
+	errorChan := make(chan ErrorData, 1)
+	outChan := make(chan []ForwardingEvent)
+
+	go func() {
+
+		batch := make([]ForwardingEvent, 0, batchSize)
+		minTime := since
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// Do nothing
+			}
+
+			err := l.CallWithTimeout(LISTFORWARDS, []interface{}{}, &reply)
+			if err != nil {
+				glog.Warningf("Error getting forwards %v\n", err)
+				if errors >= maxErrors {
+					errorChan <- ErrorData{Error: err, IsStillRunning: false}
+					return
+				}
+				errorChan <- ErrorData{Error: err, IsStillRunning: true}
+			}
+
+			for _, one := range reply.Entries {
```

Comment:
if CallWithTimeout returns error. reply.Entries might be from previous loop?

I suggest we move `var reply ClnForwardEntries` on line 458 just in case

Reply:
Solved in a different way